galsim/utilities.py

```diff
@@ -296,6 +296,25 @@ def _convertPositions(pos, units, func):
 
     return pos
 
+def _lin_approx_err(x, f, i):
+    """Error as \int abs(f(x) - approx(x)) when using ith data point to make piecewise linear
+    approximation."""
+    xleft, xright = x[:i+1], x[i:]
+    fleft, fright = f[:i+1], f[i:]
+    xi, fi = x[i], f[i]
+    mleft = (fi-f[0])/(xi-x[0])
+    mright = (f[-1]-fi)/(x[-1]-xi)
+    f2left = f[0]+mleft*(xleft-x[0])
+    f2right = fi+mright*(xright-xi)
+    return np.trapz(np.abs(fleft-f2left), xleft), np.trapz(np.abs(fright-f2right), xright)
+
+def _lin_approx_split(x, f):
+    """Optimally split a tabulated function into two-part piecewise linear approximation.
+    """
+    errs = [_lin_approx_err(x, f, i) for i in xrange(1, len(x)-1)]
```

Comment:
The Linfinity norm is indeed tough to optimize.  I haven't figured anything out on that.  But L1 and L2 would be doable.  

L1 is often called the minimax error if you do some google searching.  That's where you minimize the maximum deviation at any point, rather than the integrated absolute difference along the full path.  It seems that there are O(N) linear programming kinds of solutions for that.

L2 of course is what we would call an rms error.  It lends itself to algebraic manipulations, so there are definitely O(N) solutions to find the minimum L2 error.  

I suspect a good algorithm that combines both speed and robustness would be to use the L2 error here to find the best point to add back, but then compute the Linf error (aka the integral of the absolute deviation) in the outer loop when keeping track of the total error budget.  I suspect that the point that minimized the L2 error will often be the same or very close to the same point that would minimize the Linf error, so it should do very close to the same thing, but be significantly faster.